crates/noirc_evaluator/src/ssa/acir_gen.rs

```diff
@@ -138,6 +126,15 @@ impl Acir {
 
         let mut output = match &ins.operation {
             Operation::Binary(binary) => self.evaluate_binary(binary, ins.res_type, evaluator, ctx),
+            Operation::Constrain(value) => {
+                let mut value = self.substitute(*value, evaluator, ctx);
+                if value.witness.is_none() {
+                    value.witness = Some(generate_witness(&value, evaluator));
+                }
+                let eq_zero = from_witness(evaluate_zero_equality(&value, evaluator));
```

Comment:
There is no need to do this, the value is already 0 or 1 so we only need to check that it is one by doing something like this:
` evaluator.gates.push(subtract(&Arithmetic::one(), FieldElement::one(), &value ).into());`

Reply:
Alright, removed. I also returned `value` from the match, but perhaps I should early return from the whole function or return something else from the match instead. Not sure what the implications would be.

Reply:
A constrain does not have a value, so you could return Arithmetic::default() instead.
Whatever we return will not be used and should not matter.